File: app/javascript/channels/game_channel.js
import consumer from "channels/consumer"

consumer.subscriptions.create("GameChannel", {
  connected() {
    console.log('connected')
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    console.log('disconnected')
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    console.log('received data')
    console.log(data)
    // Called when there's incoming data on the websocket for this channel
  }
});

File: app/channels/game_channel.rb
class GameChannel < ApplicationCable::Channel
  rescue_from StandardError, with: :deliver_error_message

  # Called when the consumer has successfully
  # become a subscriber to this channel.
  def subscribed
    # stream_from "some_channel"


    # If there is a game, start a stream, else reject the subscription.
    game = Game.find(params[:id])

    if game
      stream_for game
    else
      reject
    end
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def receive(data)
    # This method is used when you rebroadcast a message.
  end
  
  private

  def deliver_error_message
    # broadcast_to(....)
  end
end
